resolwe_bio/kb/filters.py

```diff
@@ -5,11 +5,114 @@
 =======
 
 """
+from functools import reduce
+
 import django_filters as filters
+from django.contrib.postgres.search import SearchQuery, SearchRank
+from django.db import models
+from django.db.models import F
 
 from resolwe.flow.filters import TEXT_LOOKUPS
 
-from .models import Mapping
+from .models import Feature, Mapping
+
+
+class FullTextFilter(filters.BaseCSVFilter):
+    """Filter for full-text search."""
+
+    def filter(self, queryset, value):
+        """Filter field by text-search vector.
+
+        Values, given in csv format are joined with the OR operator.
+
+        Filtering by more than 135 values raises recursion error, so we have
+        to do it in chunks.
+        """
```

Comment:
Just out of curiosity: is the recursion error raised by the database or Django ORM?

Reply:
Max recursion depth is exceded in python when Django tries to resolwe the filer.